src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionService.cs

```diff
@@ -194,8 +194,11 @@ public override ElementCompletionResult GetElementCompletions(ElementCompletionC
             {
                 foreach (var completionTagName in elementCompletions.Keys)
                 {
-                    if (completionTagName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+                    if (elementCompletions[completionTagName].Count > 0 || 
```

Comment:
> Why do we need to check count > 0? I thought we were just saying catch alls wouldn't be offered as completions?

The goal of the issue is to not flood the users completion list with TagHelpers. It made sense to me to still show that catch-alls were possible on already TagHelperified elements.



> Also this is order dependent. It will only add a single one of the catch alls.

I think you're misreading the code. All CatchAlls will be applied.